crates/test-utils/src/revive.rs

```diff
@@ -0,0 +1,162 @@
+use alloy_rpc_client::ClientBuilder;
+use eyre::Result;
+use std::{process, thread::sleep, time::Duration};
+use subxt::{OnlineClient, PolkadotConfig};
+
+const NODE_BINARY: &str = "substrate-node";
+const RPC_PROXY_BINARY: &str = "eth-rpc";
+const MAX_ATTEMPTS: u32 = 15;
+const RPC_URL: &str = "http://127.0.0.1:8545";
+const RETRY_DELAY: Duration = Duration::from_secs(1);
+
+const WALLETS: [(&str, &str); 1] = [(
+    "0xf24FF3a9CF04c71Dbc94D0b566f7A27B94566cac",
+    "0x5fb92d6e98884f76de468fa3f6278f8807c48bebc13595d45af5bdc4da702133",
+)];
+
+/// Spawn and manage an instance of a compatible contracts enabled chain node.
+#[allow(dead_code)]
+struct ContractsNodeProcess {
+    node: process::Child,
+    tmp_dir: tempfile::TempDir,
+}
+
+impl Drop for ContractsNodeProcess {
+    fn drop(&mut self) {
+        self.kill()
+    }
+}
+
+impl ContractsNodeProcess {
+    async fn start() -> Result<Self> {
+        let tmp_dir = tempfile::Builder::new().prefix("cargo-contract.cli.test.node").tempdir()?;
+
+        let mut node = process::Command::new(NODE_BINARY)
+            .env("RUST_LOG", "error")
+            .arg("--dev")
+            .arg(format!("--base-path={}", tmp_dir.path().to_string_lossy()))
+            .spawn()?;
+        // wait for rpc to be initialized
+        let mut attempts = 1;
+        loop {
+            sleep(RETRY_DELAY);
```

Comment:
`tokio::time::sleep` ?

Reply:
tokio is not the crate dependency now - why do you want to change it?

Reply:
ah, missed that's it only used in tests